packages/mobile/src/qrcode/utils.ts

```diff
@@ -100,26 +101,25 @@ export function* handleBarcode(
   isOutgoingPaymentRequest?: true,
   requesterAddress?: string
 ) {
-  let data: { address: string; e164PhoneNumber: string; displayName: string } | undefined
-
+  let data: object
   try {
     data = JSON.parse(barcode.data)
   } catch (e) {
     Logger.warn(TAG, 'QR code read failed with ' + e)
-  }
-  if (typeof data !== 'object' || isEmpty(data.address)) {
-    yield put(showMessage(ErrorMessages.QR_FAILED_NO_ADDRESS))
     return
   }
-  if (!isValidAddress(data.address)) {
-    yield put(showMessage(ErrorMessages.QR_FAILED_INVALID_ADDRESS))
+
+  const either = qrDataFromJson(data)
+  if (isLeft(either)) {
```

Comment:
Is the library `fp-ts/lib/Either` used just to differentiate qrDataTypes? Could this by done just by checking the type directly/whether a particular exists?

Reply:
this emits specific parsing failures, like an invalid country code in the phone number field
I think this is a significant improvement over the legacy parsing